sod/warrior.js

```diff
@@ -0,0 +1,403 @@
+const warrior = (function () {
```

Comment:
This function hack allows the consts inside the function to be separate from global scope.